api/internal/upgrade/upgrade.go

```diff
@@ -32,6 +32,11 @@ const (
 	updaterImageEnvVar = "UPGRADE_UPDATER_IMAGE"
 )
 
+var (
+	errUpdateFailure    = errors.New("update failure")
```

Comment:
I don't see a reason why we need `errors.New`, a simple string const will be enough (we're using only the string value anyway)